From c56644479ebdad8e27e2a8d61c73aa3db9c78ca7 Mon Sep 17 00:00:00 2001
From: Tim Gover <tim.gover@raspberrypi.org>
Date: Thu, 22 Oct 2020 15:30:55 +0100
Subject: [PATCH 0258/1085] watchdog: bcm2835: Ignore params after the
 partition number

Use sscanf to extract the partition number and ignore extra parameters
which are only relevant to other reboot notifiers.
---
 drivers/watchdog/bcm2835_wdt.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/drivers/watchdog/bcm2835_wdt.c
+++ b/drivers/watchdog/bcm2835_wdt.c
@@ -127,10 +127,12 @@ static int bcm2835_restart(struct watchd
 {
 	struct bcm2835_wdt *wdt = watchdog_get_drvdata(wdog);
 
-	unsigned long long val;
+	unsigned long val;
 	u8 partition = 0;
 
-	if (data && !kstrtoull(data, 0, &val) && val <= 63)
+	// Allow extra arguments separated by spaces after
+	// the partition number.
+	if (data && sscanf(data, "%lu", &val) && val < 63)
 		partition = val;
 
 	__bcm2835_restart(wdt, partition);
